x-pack/plugins/security_solution/common/endpoint/schema/actions.ts

```diff
@@ -69,6 +73,16 @@ export const ActionDetailsRequestSchema = {
   }),
 };
 
+// TODO: fix the odd TS error
+const commandsSchema = schema.oneOf(
+  // @ts-expect-error TS2769: No overload matches this call
```

Comment:
Interesting one. I don't know which is better, but you could also cast it here:

```typescript
RESPONSE_ACTION_COMMANDS.map((command) => schema.literal(command)) as Parameters<typeof schema.oneOf>[0]
```
(same for statuses if you decide to make the change)

The issue seems to be that the `oneOf()` is typed to accept a max 10 items and a minimum of 1 item and it must know what each item in the array is, thus making it a tuple. I tried a few things with how `RESPONSE_ACTION_COMMANDS` is defined in hopes to address this TS error, but was unsuccessful. 

Typescript is not interpreting `RESPONSE_ACTION_COMMANDS` a tuple, so it complains that perhaps the array does not have enough items.

Reply:
So I tried this out and the TS didn't complain about the schema. Although, TS now interprets the properties as `Type<unknown>`  and so TS errors on all the other places where they are used.

![Screenshot 2022-09-08 at 10 11 12](https://user-images.githubusercontent.com/1849116/189070564-e3607ed8-4e80-4d9b-b69c-75a80ba6e4df.png)

I will leave this with the ts ignore for now.